src/hyperion/experiment_plans/flyscan_xray_centre_plan.py

```diff
@@ -179,7 +180,12 @@ def run_gridscan(
 
     fgs_motors = fgs_composite.fast_grid_scan
 
-    # TODO: Check topup gate
+    # Check topup gate
+    yield from check_topup_and_wait_if_necessary(
+        fgs_composite.synchrotron,
+        parameters.hyperion_params.detector_params,
+    )
+
```

Comment:
Query: could we actually delay this until a bit later in the plan?

Intuitively I might expect that we can arm the detector even if we're waiting for a topup, so that as soon as the topup finishes we're ready to start collecting.

Or if not could you add a brief comment explaining why in the code?

Reply:
Yes, the latest we can do it is just before `yield from bps.kickoff(fgs_motors)`